### PR TITLE
Return of the neck slot!

### PR DESCRIPTION
**Pull request by Bm0n**
## What Does This PR Do

Huuuge thank you to ElorgRHG and SabreML for their original PR #16346. All I did was review/modernize the code.

This is a more simplified version of their original PR, I am forgoing changing pre-existing items into neck slot ones. This is to avoid player loadout config changes in this PR.

The idea is to get this PR in as a baseline for the feature and then do other accessory items that make sense to be neck slot ones (scarfs, ties, neck bandanas, mantles, ponchos).

In a nutshell, this PR adds a neck slot. A very popular feature shared upon many SS13 servers. As an example of this feature, I have chosen to add the cloaks made by Beestations Naevi. Each head has one for their respective job and they spawn in the garment bag.

Bedsheets and the herald cloak have also been made into neck slot items. This is mainly because I used the bedsheets for testing and Qwerty requested that I make the herald cloak into a neck slot item.

## Why It's Good For The Game

A neck slot allows us to decouple uniform accessories into their own slot (when it makes sense). I personally want this which is why I have taken the time to deconflict and modernize ElorgRHG's old PR.

The ability to wear an accessory standalone and have it layer above your modsuit is cool.

## Images of changes

image

image
I think this looks way better now that the ugly top left gap is filled.

image

## Testing

- I spawned myself in-game
- Checked each command locker for the cloak
- Tested each cloak
- Checked for layering issues
- Tested the herald cloak special ability to make sure I didn't completely fuck it up

This is going to need a lot of reviewing.

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl: Bmon, ElorgRHG, SabreML
add: Added a neck slot to the game. Right now only a few items take advantage of this slot, the idea is to slowly change over items when it makes sense.
add: As an example of a neck slot item cloaks have been added to each command member's garment bag. The sprite is from Beestation contributor Naevi.
tweak: Herald cloak is now a neck slot item.
tweak: Bedsheets are now a neck slot item.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/storage/garment.dm [hunk 1]
```diff
@@ -15,6 +15,7 @@
 		/obj/item/clothing/suit/space,
 		/obj/item/clothing/mask/cigarette,
 		/obj/item/clothing/mask/facehugger, //Why would you do this
+		/obj/item/clothing/neck/cloak/,
```
Comment (Henri215) [inline code/game/objects/items/weapons/storage/garment.dm:18]:
```suggestion
		/obj/item/clothing/neck/cloak,
```

File: code/modules/clothing/neck/cloaks.dm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	icon_state = "cecloak"
+
+/obj/item/clothing/neck/cloak/research_director
+	name = "research director's cloak"
+	desc = "Worn by the leader of Scientopia, the greatest thaumaturgist and researcher of rapid unexpected self disassembly."
+	icon_state = "rdcloak"
+
+/obj/item/clothing/neck/cloak/captain
+	name = "captain's cloak"
+	desc = "Worn by the supreme leader of Space Station 13."
```
Comment (Henri215) [inline code/modules/clothing/neck/cloaks.dm:37]: game name mentioned, roll credits.

**Review by Henri215 — COMMENTED** [review]
Owl/griffin cloak is not considered a cloak?

File: code/modules/clothing/neck/cloaks.dm (hunk 2 above)
Comment (Bm0n) [reply code/modules/clothing/neck/cloaks.dm:37]: shhh it gets replaced with the station name on initialization

**Comment by Bm0n** [comment]
> griffin cloak

I believe it's a suit... unless I am wrong? I'll take a look at it tomorrow

**Comment by dj-34** [comment]
The 3rd PR is the lucky one, I believe

**Comment by ErdGinalD** [comment]
Actually, i dont want to return this, because i remember this time when ALL dudes was walk into mantles, one by one, thinking this is cool af. But in reality, you just become one of the blue/red/white point, completely override any accessory or cool clothes. If you want a mantle - so do it, but not above all shit in the world

**Comment by ErdGinalD** [comment]
Code looks good for me. Personaly against it tho.

**Review by ErdGinalD — APPROVED** [review]

File: code/modules/mob/living/carbon/human/human_update_icons.dm [hunk 3]
```diff
@@ -1051,6 +1052,26 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
 			overlays_standing[FACEMASK_LAYER] = standing
 	apply_overlay(FACEMASK_LAYER)
 
+/mob/living/carbon/human/update_inv_neck()
```
Comment (lewcc) [inline code/modules/mob/living/carbon/human/human_update_icons.dm:1055]: Please add an `update_hud_neck()`, similar to what the other update_inv_* procs have, and include it in this proc. This is necessary for observing to work properly.

**Review by lewcc — CHANGES_REQUESTED** [review]
Please make this change and verify that it works properly with observing.

File: code/modules/mob/living/carbon/human/human_update_icons.dm (hunk 3 above)
Comment (Bm0n) [reply code/modules/mob/living/carbon/human/human_update_icons.dm:1055]: Should be fixed now

**Pull request merged**